hb-06

julia-shvabovich opened the pull request.

julia-shvabovich committed:
hb-06

julia-shvabovich committed:
checkstyle

dariiabutorina commented on src/main/java/cinema/Main.java:
```diff
@@ -51,10 +52,15 @@ public static void main(String[] args) {
+        ShoppingCartService shoppingCartService
+                = (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+        shoppingCartService.registerNewShoppingCart(testUser);
+        shoppingCartService.addSession(movieSession, testUser);
```
How about using all of the service methods?

dariiabutorina commented on src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java:
```diff
@@ -0,0 +1,68 @@
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
```
Let's return an Optional from the DAO layer

Sofasmile replied:
@DariiaPikul в даному випадку це не є обов'язковим повертати Optional

dariiabutorina commented on src/main/java/cinema/model/ShoppingCart.java:
```diff
@@ -0,0 +1,48 @@
+
+    public void setUser(User user) {
+        this.user = user;
+    }
```
You're passing your entities to the exceptions' messages.
How do we know the entity details since we don't have the toString() overridden?

dariiabutorina commented on src/main/java/cinema/service/impl/ShoppingCartServiceImpl.java:
```diff
@@ -0,0 +1,48 @@
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    ShoppingCartDao shoppingCartDao;
+    @Inject
+    TicketDao ticketDao;
```
Access modifiers

julia-shvabovich replied:
no comments...

dariiabutorina commented on src/main/java/cinema/service/impl/ShoppingCartServiceImpl.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    @Override
+    public void registerNewShoppingCart(User user) {
```
How about using this method in the AuthenticationService?

Sofasmile commented on src/main/java/cinema/model/ShoppingCart.java:
```diff
@@ -0,0 +1,48 @@
+    private Long id;
+    @OneToMany
+    private List<Ticket> tickets;
+    @OneToOne
```
чому ти вирішила не використовувати тут MapsId?

julia-shvabovich replied:
done

julia-shvabovich committed:
added @MapsId in ShoppingCart;
overrode methods toString() in models;
added cart registration to AuthenticationServiceImpl;
replaced methods persist() to save();
